### ISSUE TITLE
UI: web search/Groups width not remembered

### ISSUE THREAD

**Issue by ilippert** [JabRef/jabref#8907]
### JabRef version

Other (please describe below)

### Operating system

GNU / Linux

### Details on version and operating system

JabRef 5.7--2022-06-15--cb5fe60 Linux 5.18.5-200.fc36.x86_64 amd64  Java 18.0.1  JavaFX *unknown*

### Checked with the latest development build

- [X] I made a backup of my libraries before testing the latest development version.
- [X] I have tested the latest development version and the problem persists

### Steps to reproduce the behaviour

1. change the width of the groups sidebar to your desired width
2. restart jabref
3. see that the desired width is not remembered

**Comment by ThiloteE** [comment]
JabRef 5.8--2022-08-22--a255311
Linux 5.4.0-124-generic amd64   (Linux Mint Cinnamon 20.3)
Java 18.0.2 
JavaFX 18.0.2+2
Portable Version of JabRef

Unfortunately can still reproduce.

If I remember right, in Version 5.5 or 5.6 (I would need to crosscheck) this behaviour was different but not sure.

**Comment by Siedlerchr** [comment]
Cannot reproduce in the latest development version.

**Comment by ThiloteE** [comment]
I tested again. I definitely was able to reproduce. Even reset my preferences. Maybe it is a Linux Mint Problem only...

**Comment by ThiloteE** [comment]
JabRef 5.6 works correctly. Something in between 5.6 and 5.7 broke it for me.

**Comment by Siedlerchr** [comment]
Can you still reproduce this in 5.8?

**Comment by ilippert** [comment]
JabRef 5.10--2023-01-12--5d24847
Linux 6.0.18-300.fc37.x86_64 amd64 
Java 19.0.1 
JavaFX 19+11

cannot reproduce anymore, so am closing this issue.

**Comment by ThiloteE** [comment]
@Siedlerchr yes, can still reproduce this **NOT** working in JabRef 5.9 stable on Linux Mint Cinnamon (20.3). If I remember correctly, Linux Mint uses the Nemo window/desktop manager. I still have not tried the new Linux Mint 21.xx version. I really would like to know, if a newer version behaves similar. Since width behaves correctly on Mac, Windows and Fedora, Linux Mint is the odd one out.

Interestingly, when I tried JabRef 5.5, the group width was **exactly** in the location I put it with version 5.9. So JabRef actually seems to store the location, but with 5.9 it is "overriden" with the default location, whereas in 5.5 the info from the storage is used.

This issue is not fixed, so I reopen. By fixing the issue for other versions of JabRef, it broke for Linux Mint.

Unless you want me to open a new issue for this @ilippert ?

**Comment by ilippert** [comment]
> Unless you want me to open a new issue for this @ilippert ?

no need, do keep it open here :)

**Comment by ThiloteE** [comment]
For anybody who wants to fix this: First try to reproduce with Linux Mint 21.xx

**Comment by Siedlerchr** [comment]
@ilippert  Is this still an issue with the latest development version (now with javafx20 ?) 

https://builds.jabref.org/main

**Comment by ilippert** [comment]
JabRef 5.10--2023-04-16--d47ed31
Linux 6.2.11-200.fc37.x86_64 amd64 
Java 19.0.2 
JavaFX 20+19

Issue not solved here. @ThiloteE, how about your system?

**Comment by ThiloteE** [comment]
Today I tested the newest development version of JabRef (5.10-2023-06-26) on Linux Mint 21.1
Behaviour is similar to Linux Mint 20.x

https://github.com/JabRef/jabref/issues/8907#issuecomment-1381749613 still is relevant.

**Comment by ilippert** [comment]
JabRef 5.14--2024-06-03--109600b
Linux 6.8.11-300.fc40.x86_64 amd64 
Java 21.0.2 
JavaFX 22.0.1+7

Still not fixed on Fedora 40 - sorry, https://github.com/JabRef/jabref/pull/8936 did not seem to have the desired effect.

**Comment by InAnYan** [comment]
Linux Mint 21.3 x86_64

Kernel: 5.15.0-116-generic

Desktop Environment: Xfce 4.18
Windows manager: Xfwm4

Sometimes I have this issue, sometimes not :smile:

**Comment by ryan-carpenter** [comment]
~This is resolved on my system using the current development version.~

Edit: 2025/07/09

Still an issue on
```
Operating System: openSUSE Tumbleweed 20250624
KDE Plasma Version: 6.4.0
KDE Frameworks Version: 6.15.0
Qt Version: 6.9.1
Kernel Version: 6.15.3-1-default (64-bit)
Graphics Platform: Wayland

JabRef 100.0.0
Linux 6.15.3-1-default amd64 
Java 24.0.1 
JavaFX 24.0.1+4
```

**Comment by charstnut** [comment]
Hi,

The issue still persists on:

Operating System: Kubuntu 25.04
KDE Plasma Version: 6.3.5
KDE Frameworks Version: 6.14.0
Qt Version: 6.8.3
Kernel Version: 6.14.0-23-generic (64-bit)
Graphics Platform: Wayland

even with the 6.0-alpha.2 version (also tested the 5.15 release)
JabRef 6.0-alpha.2--2025-04-27--6d0d787
Linux 6.14.0-23-generic amd64 
Java 23.0.1 
JavaFX 24+29

**Comment by koppor** [comment]
I remember that there was a PR lately, put I don't find it.

**Comment by ilippert** [comment]
> I remember that there was a PR lately, put I don't find it.

not sure what lately means ;) https://github.com/JabRef/jabref/pull/8936 is probably not what you meant, pointing to it is probably too obvious - please forgive me - but just in case, i rather share it than not :)

**Comment by koppor** [comment]
> I remember that there was a PR lately, put I don't find it.

It was PR https://github.com/JabRef/jabref/pull/13405# - fixing issue https://github.com/JabRef/jabref/issues/13402

PR still unmerged, You can try out it by following the steps at https://blog.jabref.org/2025/05/31/run-pr/

**Comment by 0xRozier** [comment]
Hi @tanvirkahlon, can you give a sign of life, or could you unassigned this issue, please ? It's been 2 weeks since you took it, and there is nothing.

**Comment by tanvirkahlon** [comment]
Go ahead @0xRozier

**Comment by 0xRozier** [comment]
Nooo, I was not fast enough, didn't know I had to be this reactive

**Comment by ThiloteE** [comment]
In this case, I would argue first come, first serve and @0xRozier was first.
@saulsgrm please choose a different [good first issue](https://github.com/JabRef/jabref/issues?q=is%3Aissue%20state%3Aopen%20label%3A%22good%20first%20issue%22). There are also[ good second issues](https://github.com/JabRef/jabref/issues?q=is%3Aissue%20state%3Aopen%20label%3A%22good%20second%20issue%22).

Maybe the following Kanban-boards will be of help too, although those are usually taken by a group of students or somebody with a little bit of experience in the code base:
- https://github.com/orgs/JabRef/projects/7/views/1
- https://github.com/orgs/JabRef/projects/3

In principle, you could work together on this particular issue, if you are both fine with it, but we try to keep good first issues only assigned to a single person, as most of our contributors are students and we want to avoid problems with grading.

**Comment by saulsgrm** [comment]
Hi @ThiloteE ,  I am now just seeing this, having already solved the bug.

I originally though that my peer @0xRozier was a core member of JabRef asking the person before me for results on this issue given that he did not explicitly request to be hand over the assignment. That is why I requested this issue. 

Here is my fork with the solution in the branch bug-8907-fix: https://github.com/saulsgrm/abbrv.jabref.org/tree/bug-8907-fix

**Comment by 0xRozier** [comment]
Thanks @ThiloteE but tt's not a big deal, I will find an other issue to work on, @saulsgrm you can keep working on this.

**Comment by koppor** [comment]
> I am extremely disappointed in the reviewers of the [PR](https://github.com/JabRef/jabref/pull/15383) on this issue.

Thank you for letting us know. This strengthens again that this project does not match your expectations — and that is completely OK. We are all humans with different desires and expectations.

I notice that your feedback focuses on what has gone wrong in the past. That is understandable, and the feelings behind it are valid. However, for JabRef to move forward, we would need proposals oriented toward the *future* — concrete ideas, contributions, or suggestions for what *should* happen. Without that, there is little we can act on.

If you are interested in the distinction, this maps well onto what NLP calls "[Towards/Away From pattern](https://nlppod.com/useful-metaprograms-in-business-towards-away-from/)": focusing on what we want to move *away from* (problems, disappointments) vs. what we want to move *toward* (goals, solutions). Both are natural, but only the latter tends to drive projects forward.

Regarding the [Code of Conduct update](https://github.com/JabRef/jabref/commit/7349649eafe867bbbe39ca5de6cfa481053c2a95): the change you refer to consists of two words. We stand by them.

We wish you well, and hope you find projects that better suit your expectations and workflow.

**Comment by Sudeep-A-Kulkarni** [comment]
Hi! I'm a GSoC 2026 applicant interested in working on this issue. I have JabRef v100.0.0 running locally on Linux and have already explored the codebase.
My proposed fix:
I traced the issue to these files:

jabgui/src/main/java/org/jabref/gui/sidepane/SidePane.java
jabgui/src/main/java/org/jabref/gui/preferences/JabRefGuiPreferences.java

The fix involves:

Reading the saved sidePaneWidth preference on startup and applying it via setDividerPositions()
Adding a listener on the divider so when the user resizes, the new width is saved via getDividerPositions()

Could a maintainer please assign this issue to me? Happy to submit a PR soon.

**Comment by Sudeep-A-Kulkarni** [comment]
> /assign-me
Hi @AnvitaPrasad! I noticed this issue is assigned to you. Are you still actively working on it? I'd be happy to help or pick it up if you've moved on. Just let me know!

**Comment by AnvitaPrasad** [comment]
Hey @Sudeep-A-Kulkarni ! I was able to reproduce the issue locally, so I went ahead and started working on it and just opened a PR (#15390).
Feel free to take a look. Happy to hear any suggestions or improvements you might have :D

**Comment by subhramit** [comment]
@AnvitaPrasad Please avoid snatching issues from one-another as it goes against our community spirit.
@Sudeep-A-Kulkarni you can help test and review the PR in the meanwhile if you wish to participate.

**Comment by AnvitaPrasad** [comment]
@subhramit Absolutely, that makes sense. I didnt intend to step on anyones toes. I saw it was unassigned after all the recent activity in the past few days and thought it was free to take up. I’ll be more careful about that next time
